cylc/uiserver/handlers.py

```diff
@@ -59,7 +59,11 @@ def initialize(self, path):
     def get(self):
         """Render the UI prototype."""
         index = os.path.join(self._static, "index.html")
-        self.write(open(index).read())
+        user = self.get_current_user()
+        protocol = self.request.protocol
+        host = self.request.host
+        base_url = f'{protocol}://{host}/{user["server"]}'
+        self.render(index, python_base_url=base_url)
```

Comment:
So this is in-effect the redirect to the UI?

What's the functional difference between `self.write` and `self.render`?

(I don't understand enough about Tornado)

Reply:
With `.write` you have to replace variables beforehand. I used it because we didn't need to replace anything in the code, no need to "render" anything.

Now with the variable, calling `.render` is simpler, as it does variable replacement for us.

Reply:
Thanks for explaining.